enhancements/installer/vsphere-ipi-zonal.md

```diff
@@ -279,24 +315,160 @@ type VSpherePlatformSpec struct {
+type VSpherePlatformFailureDomainSpec struct {
+	// regionType defines the type of region that is used for this failure domain. The current available options
+	// are ComputeCluster and Datacenter.
+	// +openshift:validation:FeatureGateAwareEnum:featureGate=VSphereHostVMGroupZonal,enum=ComputeCluster;Datacenter
+	// +kubebuilder:validation:MinLength=9
+	// +kubebuilder:validation:MaxLength=14
+	// +kubebuilder:validation:Optional
+	// +optional
+	RegionType string `json:"regionType,omitempty"`
+
+	// zoneType defines the type of zone that is used for this failure domain. The current available options
+	// are HostGroup and ComputeCluster.
+	// +kubebuilder:validation:MinLength=9
+	// +kubebuilder:validation:MaxLength=14
+	// +kubebuilder:validation:Optional
+	// +openshift:validation:FeatureGateAwareEnum:featureGate=VSphereHostVMGroupZonal,enum=HostGroup;ComputeCluster
+	// +optional
+	ZoneType string `json:"zoneType,omitempty"`
+}
+
+// VSpherePlatformTopology holds the required and optional vCenter objects - datacenter,
+// computeCluster, networks, datastore and resourcePool - to provision virtual machines.
+type VSpherePlatformTopology struct {
+	// affinity holds the VMGroup and the HostGroup names in vCenter corresponds to 
+	// a vm-host group of type Virtual Machine and Host respectively. Is also
+	// contains the VMHostRule which is an affinity vm-host rule in vCenter.
+	// +openshift:validation:featureGate=VSphereHostVMGroupZonal
+	// +kubebuilder:validation:Optional
+	// +optional
+	Affinity VSphereFailureDomainAffinity `json:"affinity,omitempty"`
+}
+ 
+// VSphereFailureDomainAffinity contains the vCenter cluster vm-host group (virtual machine and host types) 
+// and the vm-host affinity rule that together creates a affinity configuration for vm-host based zonal. 
+// This configuration within vCenter creates the required association between a failure domain, virtual machines
+// and ESXi hosts to create a vm-host based zone.
+type VSphereFailureDomainAffinity struct {
+	// vmGroup is the name of the vm-host group of type virtual machine within vCenter for this failure domain.
+	// This field is required when the VSphereFailureDomain ZoneType is HostGroup
+	// +openshift:validation:featureGate=VSphereHostVMGroupZonal
+	// +kubebuilder:validation:MaxLength=80
+	// +optional
+	VMGroup string `json:"vmGroup,omitempty"`
+
+	// hostGroup is the name of the vm-host group of type host within vCenter for this failure domain.
+	// This field is required when the VSphereFailureDomain ZoneType is HostGroup
+	// +openshift:validation:featureGate=VSphereHostVMGroupZonal
+	// +kubebuilder:validation:MaxLength=80
+	// +optional
+	HostGroup string `json:"hostGroup,omitempty"`
```

Comment:
same as above.

Reply:
The installer will not create the hostGroup, that would be kinda hard since we would need to know each esxi host per zone.